File: 2019/day2/main.go
package main

import (
	"os"
	"strings"

	"github.com/zizouhuweidi/advent-of-code/pkg/cast"
	"github.com/zizouhuweidi/advent-of-code/pkg/util"
)

func main() {
	input := util.ReadFile(os.Args[1])
	arr := strings.Split(input, ",")

	var numArr []int
	for _, v := range arr {
		numArr = append(numArr, cast.ToInt(v))
	}

	println(Part1(numArr))
}

func Part1(input []int) int {
	for i := 0; i < len(input); i += 4 {
		opcode := input[i]
		if opcode == 99 {
			break
		}
		switch opcode {
		case 1:
			input[input[i+3]] = input[input[i+1]] + input[input[i+2]]
		case 2:
			input[input[i+3]] = input[input[i+1]] * input[input[i+2]]
		default:
			continue
		}
	}

	return input[0]
}

File: 2019/day2/main.py
import sys


def Part1(s: list[int]) -> int:
    for i in range(0, len(s), 4):
        opcode = s[i]
        if opcode == 99:
            break
        if opcode == 1:
            s[s[i + 3]] = s[s[i + 1]] + s[s[i + 2]]
        elif opcode == 2:
            s[s[i + 3]] = s[s[i + 1]] * s[s[i + 2]]
    return s[0]


# def Part2()


if __name__ == "__main__":
    with open(sys.argv[1]) as f:
        s = f.read().strip().split(",")

    s = list(map(int, s))
    s[1] = 12
    s[2] = 2
    print(Part1(s))
